notification/events.go

```diff
@@ -547,18 +545,20 @@ func GetEnvForEvent(ctx context.Context, event models.Event) (*celVariables, err
 	return &env, nil
 }
 
-func shouldSilence(celEnv map[string]any, matchingSilences []models.NotificationSilence) (bool, error) {
+func shouldSilence(ctx context.Context, celEnv map[string]any, matchingSilences []models.NotificationSilence) bool {
 	for _, silence := range matchingSilences {
 		if silence.Filter != "" {
 			if ok, err := gomplate.RunTemplateBool(celEnv, gomplate.Template{Expression: string(silence.Filter)}); err != nil {
-				return false, fmt.Errorf("failed to run filter expression(%s): %w", silence.Filter, err)
+				ctx.Errorf("failed to run silence filter expression(%s): %v", silence.Filter, err)
+				logs.IfError(db.UpdateNotificationSilenceError(ctx, silence.ID.String(), err.Error()), "failed to update notification silence")
```

Comment:
this is a type safe update, so that the draft mode and module replacement are not necessary - on open the build will fail, and then the PR can just be rebased to make it pass

```suggestion
				logs.IfError(ctx.DB().Model(&filter).Select("error").Update(models.NotificationSilence{Error: lo.ToPtr(err.Error())}), "failed to update notification silence")
```